pkgs/development/libraries/libgit2-glib/default.nix

```diff
@@ -26,6 +27,14 @@ stdenv.mkDerivation rec {
     sha256 = "EzHa2oOPTh9ZGyZFnUQSajJd52LcPNJhU6Ma+9/hgZA=";
   };
 
+  patches = [
+    (fetchpatch2 {
+      name = "support-libgit2-1.8.patch";
+      url = "https://github.com/GNOME/libgit2-glib/pull/4/commits/304a8c3fae1784d665e92c1ce2143c66d7ecf47f.patch";
```

Comment:
This patch looks insufficient compared to https://gitlab.gnome.org/GNOME/libgit2-glib/-/merge_requests/40

Also it is unmerged so we are not certain the URL will persist when/if the commit garbage collected.

And `fetchpatch2` should not be used for dynamically-generated patches – it produces unstable FODs because short git commit ids in the patch will be lengthened eventually.

Reply:
> This patch looks insufficient

it builds as is

> if the commit garbage collected

gh doesn't gc ever, unless you file a support ticket

> And `fetchpatch2` should not be used for dynamically-generated patches – it produces unstable FODs because short git commit ids in the patch will be lengthened eventually.

Not sure what you're asking for here

Reply:
> it builds as is

It lacks the casts, which I suspect might be necessary on some configurations (maybe CLang).

> gh doesn't gc ever, unless you file a support ticket

Is that true of unmerged PRs? I have seen some such PRs, whose content has disappeared (though it was a long time ago).

> Not sure what you're asking for here

`fetchpath` should be used instead of `fetchpatch2` when fetching dynamically generated patch. `fetchpatch2` is unfortunately only suitable for transforming static patches

Reply:
> Is that true of unmerged PRs? I have seen some such PRs, whose content has disappeared (though it was a long time ago).

I think they can if a PR is force pushed and maybe never accessed but that's GitLab.

fetchpatch2 is required for renames. Also I never heard of those issues. Some people are using fetchpatch2 by default to get it into the tree more.

Reply:
https://github.com/NixOS/nixpkgs/issues/257446

Reply:
I've changed the url to use the actual commit and added `full_index=1` as a param to forcibly use the full hash in the patch itself; this should obviate concerns about the patch being dynamic now.